messaging/src/main/java/org/axonframework/queryhandling/SimpleQueryHandlingComponent.java

```diff
@@ -17,62 +17,106 @@
  * @author Steven van Beelen
  * @since 5.0.0
  */
-public class SimpleQueryHandlingComponent
-        implements QueryHandlingComponent, QueryHandlerRegistry<SimpleQueryHandlingComponent> {
+public class SimpleQueryHandlingComponent implements
+        QueryHandlingComponent,
+        QueryHandlerRegistry<SimpleQueryHandlingComponent>,
+        DescribableComponent {
 
-    private final ConcurrentHashMap<QueryHandlerName, QueryHandler> queryHandlers;
+    private final String name;
+    private final Map<QueryHandlerName, QueryHandler> queryHandlers = new HashMap<>();
+    private final Set<QueryHandlingComponent> subComponents = new HashSet<>();
 
-    public SimpleQueryHandlingComponent() {
-        this.queryHandlers = new ConcurrentHashMap<>();
+    /**
+     * Instantiates a simple {@link QueryHandlingComponent} that is able to handle query and delegate them to
+     * subcomponents.
+     *
+     * @param name The name of the component, used for {@link DescribableComponent describing} the component.
+     * @return A simple {@link QueryHandlingComponent} instance with the given {@code name}.
+     */
+    public static SimpleQueryHandlingComponent create(@Nonnull String name) {
+        return new SimpleQueryHandlingComponent(name);
+    }
+
+    private SimpleQueryHandlingComponent(@Nonnull String name) {
+        this.name = Assert.nonEmpty(name, "The name may not be null or empty.");
+    }
+
+    @Override
+    public SimpleQueryHandlingComponent subscribe(@Nonnull QueryHandlerName queryName,
+                                                  @Nonnull QueryHandler handler) {
+        queryHandlers.put(queryName, handler);
+        return this;
+    }
+
+    @Override
+    public SimpleQueryHandlingComponent subscribe(@Nonnull QueryHandlingComponent handlingComponent) {
+        subComponents.add(handlingComponent);
+        return this;
     }
 
     @Nonnull
     @Override
     public MessageStream<QueryResponseMessage> handle(@Nonnull QueryMessage query,
                                                       @Nonnull ProcessingContext context) {
         QualifiedName name = query.type().qualifiedName();
+        // TODO get QualifiedName as a responseType i.o. making one here for the response
         QueryHandlerName handlerName = new QueryHandlerName(
                 name, new QualifiedName(query.responseType().getExpectedResponseType())
         );
-        QueryHandler handler = queryHandlers.get(handlerName);
-        if (handler == null) {
-            // TODO #3488 this would benefit from a dedicate exception
-            return MessageStream.failed(new IllegalArgumentException(
-                    "No handler found for query with name [" + name + "]"
-            ));
+        Optional<QueryHandlingComponent> optionalSubHandler =
+                subComponents.stream()
+                             .filter(subComponent -> subComponent.supportedQueries().contains(handlerName))
+                             .findFirst();
```

Comment:
What if many `subComponents` can handle same `QueryHandlerName`? Should it be prevented?